internal/converter/embdemd_openapi_converter_util.go

```diff
@@ -604,3 +604,337 @@ func MapEmbedMDPropertyModelVersionIdServeModel(source *[]models.Properties) (st
+	for _, v := range *source {
+		if v.Name == "status" {
+			if v.StringValue == nil {
+				return nil, fmt.Errorf("%w: status is required", api.ErrBadRequest)
+			}
+
+			experimentRunStatus, err := openapi.NewExperimentRunStatusFromValue(*v.StringValue)
+			if err != nil {
+				return nil, err
+			}
+
+			return experimentRunStatus, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func MapEmbedMDPropertyStartTimeSinceEpochExperimentRun(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "start_time_since_epoch" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertyEndTimeSinceEpochExperimentRun(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "end_time_since_epoch" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertyExperimentIdExperimentRun(source *[]models.Properties) (string, error) {
+	for _, v := range *source {
+		if v.Name == "experiment_id" {
+			result := Int32ToString(v.IntValue)
+			if result == nil {
+				return "", fmt.Errorf("experiment id is required")
+			}
+			return *result, nil
+		}
+	}
+
+	return "", fmt.Errorf("experiment id is required")
+}
+
+func MapEmbedMDExternalIDExperimentRun(source *models.ExperimentRunAttributes) *string {
+	return source.ExternalID
+}
+
+func MapEmbedMDNameExperimentRun(source *models.ExperimentRunAttributes) *string {
+	return MapNameFromOwned(source.Name)
+}
+
+func MapEmbedMDCreateTimeSinceEpochExperimentRun(source *models.ExperimentRunAttributes) *string {
+	return Int64ToString(source.CreateTimeSinceEpoch)
+}
+
+func MapEmbedMDLastUpdateTimeSinceEpochExperimentRun(source *models.ExperimentRunAttributes) *string {
+	return Int64ToString(source.LastUpdateTimeSinceEpoch)
+}
+
+// DataSet property mapping functions
+func MapEmbedMDPropertyDigest(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "digest" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertySourceType(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "source_type" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertySource(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "source" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertySchema(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "schema" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+func MapEmbedMDPropertyProfile(source *[]models.Properties) *string {
+	for _, v := range *source {
+		if v.Name == "profile" {
+			return v.StringValue
+		}
+	}
+
+	return nil
+}
+
+// DataSet mapping functions
+func MapEmbedMDURIDataSet(source *models.DataSetAttributes) *string {
+	return source.URI
+}
+
+func MapEmbedMDArtifactTypeDataSet(source *models.DataSetAttributes) *string {
+	return of("dataset")
```

Comment:
The OpenAPI schema is expecting either "DataSet" or "dataset-artifact" type and here is it wrong.

```suggestion
	return of("dataset-artifact")
```

This might also need some changes for tests?